playbooks/group_vars/all.yml

```diff
@@ -140,8 +140,8 @@ es_group: elasticsearch
 es_data_dir: "{{ rock_data_dir }}/elasticsearch"
 es_cluster_name: rocknsm
 es_node_name: "{{ ansible_hostname }}"
-es_network_host: "{{ '_site:ipv4_' if ( groups['elasticsearch'] | length ) > 1 else '_local:ipv4_' }}"
-es_url: "http://{{ groups['elasticsearch'][0] if ( groups['elasticsearch'] | length ) > 1 else '127.0.0.1' }}:9200"
+es_network_host: "{{ '_site:ipv4_' if (groups['elasticsearch']|union(groups['logstash'])|union(groups['kibana']) | length ) > 1 else '_local:ipv4_' }}"
+es_url: "http://{{ groups['elasticsearch'][0] }}:9200"
```

Comment:
if `es_network_host` is `_local:ipv4_`, then this `es_url` will only work if your ansible inventory uses the loopback. I like what you did with `es_network_host` though.